src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
 Programmable Physical Environments
 ==================================
 
+Overview
+--------
+
+Physical things includes nearly everything that will appear in life of human beings. When it comes to programmable physical things, the term should refer to a kind of physical system which might be a combination of software and hardware which can respond to the analog world. When we say 'environment', we focus more on the interactive function of this physical system with human beings. Now we can derive a definition for programmable physical environment, which is an interactively programmable physical system that can sense and respond to the real world.
+
+Programmable physical environment has a broad definition and it is used in various fields. The usage included educational use, virtual/augmented reality, factory manufacture, internet of things, building architecture, etc. The term still has a long way for improvement. In the future, programmable physical environment may become more common in the society as a combination of various technology systems.
+
+Embodied Cognition
+------------------
+
+One related theory to programmable physical environment is the embodied cognition. Embodied cognition is the theory that many features of cognition are shaped by aspects of the entire body of the organism. High level mental constructs and performance on various cognitive tasks are all included in the features of cognition. It points out that the cognition of beings is related to the entire aspects of the body. When coming to programmable physical environment, it should remind the designers in what ways can human beings interact with the systems.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "wikiembodiedcognition"
+
+  This wiki page gives a explanation of embodied cognition
+
+Design Insight
+--------------
+
+As a programmable physical environment, the human interation with the system is very important. How human will interact with the system should also be a research topic of programmable physical environment.
```

Comment:
Suggestion: "Programmable physical environments emphasize the physicality of programming: programmers interact with physical devices, and programs ultimately affect the physical world. Several authors have remarked on the need for a more expansive view of programming that accounts for and takes advantage of this physicality."